### ISSUE TITLE
Undefined names used as decorators on top of `TypedDict` are not reported

### ISSUE THREAD

**Issue by bzoracler** [python/mypy#21030]
The following shows no errors (see [mypy Playground](https://mypy-play.net/?mypy=master&python=3.12&gist=f9a7d7984ae6124654700cfe7db7d5d7)):

```python
from typing import TypedDict

@abc
@efg.hij
@klm[nop]
@qrs.tuv[wxy]
class A(TypedDict):
    pass
```

This only seems to occur on decorator expressions which also look like annotation expressions.

- Mypy version used: 1.19.1, master
